### ISSUE TITLE
Turn the benchmark back to life (proper way)

### ISSUE THREAD

**Issue by elhimov** [tarantool/tpcc#10]
Long time ago at the time when SQL support was introduced in tarantool this benchmark was adapted for tarantool, but since then it hadn't been maintained and now it doesn't work. The main problem is that there are several fields of type _decimal_ in the tables used in the benchmark, values for them are encoded as _double_ by connector and then server raises the type mismatch errors.

In order to quickly turn the benchmark into working state a kind of simple and quick solution was suggested recently (#9). It just changes the type of the mentioned fields from _decimal_ to _double_, so "type mismatches" are gone, but it looks like it is acceptable only as a temporary solution. According to the [specification](https://tpc.org/tpc_documents_current_versions/pdf/tpc-c_v5.11.0.pdf) the mentioned fields should be of _decimal_ type. Implementation over _double_ type might produce incorrect result due to the possible rounding (please, see [here](https://github.com/tarantool/tpcc/pull/7#discussion_r542201890)).

According to [this](https://github.com/orgs/tarantool/discussions/5910#discussioncomment-1016183) comment it seems that implicit conversion double->decimal is not applied always (@ImeevMA, would you, please, confirm), so it looks like the proper way to fix the problem would be changing the way these values are passed to server, that is they should be encoded as decimals, not as doubles.

For now encoding of decimals is not supported in c-connector, so first we should implement it in the connector. Decimal is the one of tarantool extensions for MessagePack which is c-implemented and integrated into core. There is no separate library for it, so to avoid code duplication first we need to rearrange somehow MessagePack extensions code to use it both in Tarantool's core and in c-connector.

**Comment by ImeevMA** [comment]
> According to [this](https://github.com/orgs/tarantool/discussions/5910#discussioncomment-1016183) comment it seems that implicit conversion double->decimal is not applied always

Indeed, sometimes the DOUBLE value may not be converted to DECIMAL, for example:
```
tarantool> box.execute([[create table t(i dec primary key);]])
---
- row_count: 1
...

tarantool> box.execute([[insert into t values(1 / 1.1e0);]])
---
- null
- 'Type mismatch: can not convert double(0.909090909090909) to decimal'
...
```
